yt_dlp/extractor/nekohacker.py

```diff
@@ -0,0 +1,125 @@
+                    'id': '03-Nature-Talk',
+                    'ext': 'mp3',
+                    'title': 'Nature Talk',
+                    'thumbnail': 'https://nekohacker.com/wp-content/uploads/2022/11/Nekoverse_Artwork-1024x1024.jpg',
+                    'vcodec': 'none',
+                    'acodec': 'mp3',
+                    'upload_date': '20221101'
+                }
+            },
+            {
+                'url': 'https://nekohacker.com/wp-content/uploads/2022/11/04-Crystal-World.mp3',
+                'md5': 'd8e59a48061764e50d92386a294abd50',
+                'info_dict': {
+                    'id': '04-Crystal-World',
+                    'ext': 'mp3',
+                    'title': 'Crystal World',
+                    'thumbnail': 'https://nekohacker.com/wp-content/uploads/2022/11/Nekoverse_Artwork-1024x1024.jpg',
+                    'vcodec': 'none',
+                    'acodec': 'mp3',
+                    'upload_date': '20221101'
+                }
+            }
+        ]
+    }]
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, playlist_id)
+
+        webpage = get_element_by_class('playlist', webpage)
+
+        if webpage is None:
+            raise ExtractorError('no playlist element found - likely not a album', expected=True)
+
+        # the h3 is acutally empty when downloaded by youtube-dl
+        # playlist_title = get_element_by_class('sr_it-playlist-title', webpage)
+        playlist_title = None
+
+        entries = []
+
+        for track in re.findall(
+                r'(<li[^>][^>]+>)',
+                webpage):
+            attr = extract_attributes(track)
+            tracktitle = attr['data-tracktitle']
+            trackurl = attr['data-audiopath']
+
+            if playlist_title is None:
+                playlist_title = attr['data-albumtitle']
+
+            id = self._search_regex(self._ENTRY_URL, trackurl, 'trackurl id')
+
+            # full_response = self._request_webpage(trackurl, id, headers={
+            #     'Accept-Encoding': 'identity',
+            # })
+            # content_type = full_response.headers.get('Content-Type', '').lower()
+
+            upload_date = datetime.strptime(attr['data-releasedate'], '%Y.%m.%d Release')
+
+            entry = {
+                'id': id,
+                'title': tracktitle,
+                'url': trackurl,
+                'ext': determine_ext(trackurl),
+                'thumbnail': attr['data-albumart'],
+                'upload_date': unified_strdate(upload_date.strftime('%Y%m%d'))
+            }
+
+            # if re.match(r'audio/', content_type):
+            #     entry['vcodec'] = 'none'
+            if entry['ext'] == 'mp3':
+                entry['vcodec'] = 'none'
+                entry['acodec'] = 'mp3'
+
+            entries.append(entry)
+
+        return self.playlist_result(entries, playlist_id, playlist_title)
```

Comment:
This can be simplified and made safer with `traverse_obj`

 - `upload_date = datetime.strptime(attr['data-releasedate'], '%Y.%m.%d Release')` -- this is fatal and should not be, and it throws on other albums (again see https://nekohacker.com/susume/)
 - I think `release_date` is more accurate here than `upload_date`
 - `entry['vcodec'] = 'none'` -- going by the URL attribute name `data-audiopath`, I think this can always be `'none'`, yeah?
 - need to add some imports from utils (and remove some unused imports)

```suggestion
        entries = []
        for track in re.findall(r'(<li[^>]+data-audiopath[^>]+>)', playlist):
            entry = traverse_obj(extract_attributes(track), {
                'url': ('data-audiopath', {url_or_none}),
                'ext': ('data-audiopath', {determine_ext}),
                'id': 'data-trackid',
                'title': 'data-tracktitle',
                'album': 'data-albumtitle',
                'release_date': ('data-releasedate', {lambda x: re.match(r'\d{8}', x.replace('.', ''))}, 0),
                'thumbnail': ('data-albumart', {url_or_none}),
            })
            entry['vcodec'] = 'none'
            entry['acodec'] = 'mp3' if entry['ext'] == 'mp3' else None
            entries.append(entry)

        return self.playlist_result(entries, playlist_id, traverse_obj(entries, (0, 'album')))
```
&nbsp;
Or, it looks like we could also get some more metadata if we want to:
```suggestion
        entries = []
        for track_number, track in enumerate(re.findall(r'(<li[^>]+data-audiopath[^>]+>)', playlist), 1):
            entry = traverse_obj(extract_attributes(track), {
                'url': ('data-audiopath', {url_or_none}),
                'ext': ('data-audiopath', {determine_ext}),
                'id': 'data-trackid',
                'title': 'data-tracktitle',
                'track': 'data-tracktitle',
                'album': 'data-albumtitle',
                'duration': ('data-tracktime', {parse_duration}),
                'release_date': ('data-releasedate', {lambda x: re.match(r'\d{8}', x.replace('.', ''))}, 0),
                'thumbnail': ('data-albumart', {url_or_none}),
            })
            entry['track_number'] = track_number
            entry['artist'] = 'Nekohacker'
            entry['vcodec'] = 'none'
            entry['acodec'] = 'mp3' if entry['ext'] == 'mp3' else None
            entries.append(entry)

        return self.playlist_result(entries, playlist_id, traverse_obj(entries, (0, 'album')))
```

Reply:
done with b2ddc17b722bd4d8bb2cd57e50523ae028f9d86d (manually applied for imports and because github said outdated diff), using the "more metadata" approach